### PR TITLE
feat(minidump): Add processing options

### PR DESCRIPTION
**Pull request by loewenheim**
This is a proposal for making the changes in https://github.com/getsentry/symbolic/pull/461 configurable.
* `process_minidump` is renamed to `process_minidump_internal` and gains 3 new parameters `max_threads`, `max_memory_regions`, `max_modules` that are used to set the respective limits in Breakpad. Defaults to current Breakpad values.
* A new `ProcessMinindumpOptions` struct that bundles `max_{threads,memory_regions,modules}`.
* A new free function `process_minidump` that has the same signature as `ProcessState::from_minidump` plus a `ProcessMinidumpOptions` parameter.
* `ProcessState::from_minidump` just calls the new `process_minidump` with default options.

The end result should be that you can use `ProcessState::from_minidump` exactly like before, but you can also use `process_minidump` if you want control over the different limits. API wise this is probably confusing; personally I would be in favor of removing `ProcessState::from_minidump` in favor of a free function (obviously a breaking change), but we could also do the reverse and add another associated function to `ProcessState` or just change the signature of `from_minidump` if we don't think a version without the `options` parameter needs to exist.

### CODE REVIEW

File: symbolic-minidump/src/processor.rs [hunk 1]
```diff
@@ -1206,6 +1186,65 @@ impl<'a> fmt::Debug for ProcessState<'a> {
     }
 }
 
+/// Processes a minidump supplied via raw binary data.
+////// Returns a `ProcessState` that contains information about the crashed
+/// process. The parameter `frame_infos` expects a map of Breakpad symbols
+/// containing STACK CFI and STACK WIN records to allow stackwalking with
+/// omitted frame pointers.
+pub fn process_minidump<'a>(
```
Comment (flub) [inline symbolic-minidump/src/processor.rs:1194]: I wonder if API-wise this makes more sense as a new method on `ProcessState`?  Maybe `ProcessState::from_minidump_with_options`?

**Comment by flub** [comment]
nice!

**Comment by Gankra** [comment]
Presumably this is functionality you would want in a hypothetical rust-minidump backend too, right? (And if so, do you have preferred behaviour when these limits are exceeded? hard fail? silently truncate? truncate but indicate what the "true" size was supposed to be?)

(Also data point: The old mozilla mdsw included truncation on backtrace lengths (and noted the "true" size), but we ended up dropping it because users just found it confusing.)

**Comment by loewenheim** [comment]
@Gankra We are currently debating internall whether we even want to support this in the case of breakpad. I'll keep you posted.

File: symbolic-minidump/src/processor.rs [hunk 2]
```diff
@@ -1031,6 +1034,34 @@ struct SymbolEntry {
 /// [`CodeModule`]: struct.CodeModule.html
 pub type FrameInfoMap<'a> = BTreeMap<CodeModuleId, CfiCache<'a>>;
 
+/// Options passed to breakpad when processing a minidump.
+#[derive(Clone, Debug)]
+#[non_exhaustive]
+pub struct ProcessMinidumpOptions {
+    /// The maximum number of threads the minidump may contain.
+    ///
+    /// Defaults to 4096.
```
Comment (Swatinem) [inline symbolic-minidump/src/processor.rs:1043]: I think thats a lot higher already than the default in breakpad. Or do we intentionally increase all these?

**Review by Swatinem — APPROVED** [review]

**Comment by relaxolotl** [comment]
Adding some context: This has been closed as we're no longer using breakpad to parse minidumps. However, I think the question of whether we want to include these configurable toggles for the new minidump parsing backend (rust-minidump) may still be up in the air; I've opened up #551 to track this.

**Pull request closed without merging**